chore: refactor "create project" e2e tests

ZeRego opened the pull request:
Changes:
- refactor part of the test that was testing results
  - replaced UI tests with API tests
- improve selectors

Results when creating postgres project locally:

Before - 1min
<img width="401" alt="Screenshot 2023-07-07 at 13 58 34" src="https://github.com/lightdash/lightdash/assets/9117144/a2e6af0a-78d9-47cb-8d3e-f54f6af198b2">

After - 30sec
<img width="604" alt="Screenshot 2023-07-07 at 15 13 42" src="https://github.com/lightdash/lightdash/assets/9117144/6d97b2ef-c47d-4234-b543-8e81e2c9c057">


Results when creating projects in github action:

Before - 5min 28sec
<img width="1288" alt="Screenshot 2023-07-07 at 13 57 59" src="https://github.com/lightdash/lightdash/assets/9117144/c1ea8e2d-4c90-44c2-af88-afe96d305284">

After - 3min 48sec
<img width="984" alt="Screenshot 2023-07-07 at 15 52 14" src="https://github.com/lightdash/lightdash/assets/9117144/91c3185e-d3c7-42ab-8e11-c85911280104">

rephus reviewed (approved):
Looks good, we could also consider creating these projects on the CLI , perhaps this way is faster than using the UI. 

testQuery and testRunQuery already have API tests, right ?

rephus commented on packages/e2e/cypress/e2e/createProjects.cy.ts:
```diff
@@ -55,85 +57,99 @@ const warehouseConfig = {
 };
 
-const testCompile = () => {
+const testCompile = (): Cypress.Chainable<string> => {
```
What's the advantage of using this chainable ?

ZeRego replied:
So we can chain other actions and pass the project uuid 

```
        testCompile().then((projectUuid) => {
            testFilterStringEscaping(projectUuid);
            testTimeIntervalsResults(projectUuid);
            testPercentile(projectUuid);
        });
```

Note that cypress actions are not Promises so I couldn't simply return the string.

ZeRego commented:
> Looks good, we could also consider creating these projects on the CLI , perhaps this way is faster than using the UI.
> testQuery and testRunQuery already have API tests, right ?

We want to test that part in the UI since it is a core functionality. 
I removed some parts and refactored other parts to be via the API since we already have tests for the explore page.
